tests/conftest.py

```diff
@@ -64,6 +64,15 @@ def reset_tracing():
     IPythonTraceDisplayHandler._instance = None
 
 
+@pytest.fixture(autouse=True)
+def validate_trace_finish():
+    yield
+
+    # Validate all spans are finished by the end of the each test.
+    # Leaked span is critical problem and also hard to find without an explicit check.
+    assert mlflow.get_current_active_span() is None
```

Comment:
what would pytest log look like when this check fails? Is it debuggable?

Reply:
```suggestion
    span = mlflow.get_current_active_span()
    assert span is None, f"Found leaked span: {span}"
```

not sure if this is more debuggable.

Reply:
When a test leaks the span, test result will look like this:

<img width="500" alt="Screenshot 2024-11-20 at 14 17 40" src="https://github.com/user-attachments/assets/50325a0d-0173-4cee-b983-9325d59734a8">

The test case leaked the span failed and some other tests afterwards would fail due to the polluted context.

<img width="500" alt="Screenshot 2024-11-20 at 14 18 45" src="https://github.com/user-attachments/assets/3a6f099e-8aa2-49db-9602-9db0594d306c">

While the former one is kinda clear (at least tells the span is leaked from this test), but the latter one is not. To resolve it, maybe we can add a similar check at the beginning of the test, then fail with a special message like `Active span found at the beginning of the test. Span context might be polluted by some other tests. Check the test log to see if there is any test failed with leaked span validation.`. WDYT?

Reply:
> a special message like Active span found at the beginning of the test. Span context might be polluted by some other tests. Check the test log to see if there is any test failed with leaked span validation.. WDYT?

sg!